core/app/models/spree/store_credit_category.rb

```diff
@@ -1,6 +1,6 @@
 class Spree::StoreCreditCategory < Spree::Base
-  class_attribute :non_expiring_credit_types
-  self.non_expiring_credit_types = [Spree.t("store_credit.non_expiring")]
+  class_attribute :non_expiring_credit_categories
```

Comment:
I changed the naming on this because it was confusing. We have both a `Spree::StoreCreditCategory` and a `Spree::StoreCreditType` model.

This attribute is called `non_expiring_credit_types`, but its value is set to the default `Spree::StoreCreditCategory` name, not a `Spree::StoreCreditType`. Also in the `non_expiring?` method, it checks if the current `CreditCategory`'s name is in the array, so calling it a credit type is misleading.

I found this confusing while working on it, but if you prefer to keep it as is please let me know. Otherwise I'll add deprecation warnings.